app/screens/channel/channel_nav_bar/channel_drawer_button.js

```diff
@@ -62,10 +62,8 @@ class ChannelDrawerButton extends PureComponent {
         EventEmitter.on('drawer_opacity', this.setOpacity);
     }
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.mentionCount !== this.props.mentionCount) {
-            PushNotifications.setApplicationIconBadgeNumber(this.props.mentionCount);
-        }
+    componentDidUpdate() {
+        PushNotifications.setApplicationIconBadgeNumber(this.props.mentionCount);
```

Comment:
@enahum Can you elaborate on the change here? Why are we updating when the value is not changed? and should we be setting the value on mount as well here?

Reply:
Sure.

1. Push notification is received while the app is in the background and sets the badge icon.
2. When the app is opened through the icon and not the notification the channel is marked as read meaning no more unread mentions, this component updates and then it will set the right badge number in the icon.
3. If the app is opened through the notification the OS decreases the badge number, the channel is marked as read and the badge is also set once more but without changes.

The reason we do it like this is cause we want to ensure the badge is set even if the previous value and the next values are equal, the reason is this:
App is in the background and the value for mentionCount is 0, you get a notification and then you bring the app to the foreground, websocket reconnects and gets the new posts and marks the channel as read, meaning that the mentionCount is still 0 at this point but the badge number is still set to the number of push notifications received, so by doing this the badge number is set to the right number.

Makes sense now?

Reply:
Makes sense. Thanks for elaborating 👍